packages/azure-http-specs/specs/azure/resource-manager/common-properties/error.tsp

```diff
@@ -0,0 +1,159 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+import "@typespec/spector";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.ResourceManager;
+using TypeSpec.OpenAPI;
+using Spector;
+
+namespace Azure.ResourceManager.CommonProperties;
+
+@resource("confidentialResources")
+model ConfidentialResource is TrackedResource<ConfidentialResourceProperties> {
+  ...ResourceNameParameter<ConfidentialResource>;
+}
+
+@doc("Confidential Resource Properties.")
+model ConfidentialResourceProperties {
+  @visibility(Lifecycle.Read)
+  @doc("The status of the last operation.")
+  provisioningState: string;
+
+  username: string;
+}
+
+/**
+ * Api error.
+ */
+model ApiError {
+  /**
+   * The Api error details
+   */
+  details?: ApiErrorBase[];
+
+  /**
+   * The Api inner error
+   */
+  innererror?: InnerError;
+
+  /**
+   * The error code.
+   */
+  code?: string;
+
+  /**
+   * The target of the particular error.
+   */
+  target?: string;
+
+  /**
+   * The error message.
+   */
+  message?: string;
+}
+
+/**
+ * Api error base.
+ */
+model ApiErrorBase {
+  /**
+   * The error code.
+   */
+  code?: string;
+
+  /**
+   * The target of the particular error.
+   */
+  target?: string;
+
+  /**
+   * The error message.
+   */
+  message?: string;
+}
+
+/**
+ * Inner error details.
+ */
+model InnerError {
+  /**
+   * The exception type.
+   */
+  exceptiontype?: string;
+
+  /**
+   * The internal error message or exception dump.
+   */
+  errordetail?: string;
+}
+
+/**
+ * An error response.
+ */
+@error
+model CloudError {
+  /**
+   * Api error.
+   */
+  error?: ApiError;
+}
+
+@armResourceOperations
+interface Error {
+  @scenario
+  @scenarioDoc("""
+    Resource GET operation.
+    Expected path: /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.ResourceManager.CommonProperties/confidentialResources/confidential",
+    Expected query parameter: api-version=2023-12-01-preview
+    
+    Expected response status code: 404
+    Expected response body:
+    ```json
+    {
+      "error": {
+        "code": "ResourceNotFound",
+        "message": "The Resource 'Azure.ResourceManager.CommonProperties/confidentialResources/confidential' under resource group 'test-rg' was not found."
+      }
+    }
+    ```
+    """)
+  get is ArmResourceRead<ConfidentialResource>;
```

Comment:
this get can't set error model?

Reply:
Yes, it can. Here it's for testing predefined `Error` model, in contrast with user-defined `CloudError`.

Reply:
since the mock always return build-in error. it's better to rename the op.